WooCommerce/src/androidTest/kotlin/com/woocommerce/android/e2e/screens/products/ProductFilterScreen.kt

```diff
@@ -0,0 +1,65 @@
+package com.woocommerce.android.e2e.screens.products
+
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.matcher.ViewMatchers
+import com.woocommerce.android.R
+import com.woocommerce.android.e2e.helpers.util.Screen
+import org.hamcrest.Matchers
+
+class ProductFilterScreen : Screen {
+
+    companion object {
+        const val TOOLBAR = R.id.toolbar
+    }
+
+    constructor() : super(TOOLBAR)
+
+    fun filterByPropertyAndValue(property: String, value: String): ProductFilterScreen {
+        val filterItem = Espresso.onView(
+            Matchers.allOf(
+                ViewMatchers.withId(R.id.filterItemName),
+                ViewMatchers.withText(property),
+            )
+        )
+
+        clickOn(filterItem)
+
+        val filterOption = Espresso.onView(
+            Matchers.allOf(
+                ViewMatchers.withId(R.id.filterOptionItem_name),
+                ViewMatchers.withText(value),
+            )
+        )
+
+        clickOn(filterOption)
```

Comment:
depending on how many times this is going to be used again, i think there's a potential to DRY it (in `Screen.kt`). 

```suggestion
fun findAndClick(id: String, text: String) {
        val element = Espresso.onView(
            Matchers.allOf(
                ViewMatchers.withId(id),
                ViewMatchers.withText(text),
            )
        )

        clickOn(element)
}
```

Reply:
Nice! I was in a hurry somewhat, and totally missed this, thank you! I extracted this part into `Screen.clickByTextAndId()` with ee70239120c78afae1a6cbd5e1c6629c802327ff 🙇